client/blocks/inline-help/inline-help-search-results.jsx

```diff
@@ -0,0 +1,186 @@
+				title: this.props.translate( 'Get Started' ),
+				description: this.props.translate(
+					'No matter what kind of site you want to build, our five-step checklists will get you set up and ready to publish.'
+				),
+			},
+			{
+				link: 'https://en.support.wordpress.com/settings/privacy-settings/',
+				title: this.props.translate( 'Privacy Settings' ),
+				description: this.props.translate(
+					'Limit your site’s visibility or make it completely private.'
+				),
+			},
+		];
+		return helpfulResults;
+	}
+
+	renderSearchResults() {
+		if ( isEmpty( this.props.searchQuery ) ) {
+			// not searching
+			return this.renderContextHelp();
+		}
+
+		if ( this.props.isSearching ) {
+			// search, but no results so far
+			return (
+				<PlaceholderLines />
+			);
+		}
+
+		if ( isEmpty( this.props.searchResults ) ) {
+			// search done, but nothing found
+			return (
+				<div>
+					<p className="inline-help__empty-results">No results.</p>
+					{ this.renderContextHelp() }
+				</div>
+			);
+		}
+
+		// found something
+		const links = this.props.searchResults;
+		return (
+			<ul className="inline-help__results-list">{ links && links.map( this.renderHelpLink ) }</ul>
+		);
+	}
+
+	renderContextHelp() {
+		const links = this.getContextResults();
+		return (
+			<ul className="inline-help__results-list">
+				{ links && links.map( this.renderHelpLink ) }
+			</ul>
+		);
+	}
+
+	getSelectedUrl = () => {
+		if ( this.props.selectedResult === -1 ) {
+			return false;
+		}
+
+		const links = this.props.searchResults || this.getContextResults();
+		const selectedLink = links[ this.props.selectedResult ];
+		if ( ! selectedLink || ! selectedLink.link ) {
+			return false;
+		}
+		return selectedLink.link;
+	}
+
+	componentWillUpdate( nextProps ) {
+		if ( nextProps.shouldOpenSelectedResult ) {
+			const url = this.getSelectedUrl();
+			this.followHelpLink( url );
+			this.props.didOpenResult();
+			window.location = url;
+			return;
+		}
+	}
+
+	followHelpLink = ( url ) => {
+		const payload = {
+			searchQuery: this.props.searchQuery,
+			currentUrl: window.location.href,
+			resultUrl: url,
+		};
+		return () => {
+			tracks.recordEvent( 'calypso_inline-help_follow-link', payload );
+		};
+	}
+
+	renderHelpLink = ( link, index ) => {
+		const classes = [
+			'inline-help__results-item',
+			this.props.selectedResult === index && 'is-selected',
+		].filter( Boolean );
+		return (
+			<li key={ link.link } className={ classNames( ...classes ) }>
+				<a
+					href={ link.link }
+					onClick={ this.followHelpLink( link.link ) }
+					title={ decodeEntities( link.description ) }
+				>
+					{ preventWidows( decodeEntities( link.title ) ) }
+				</a>
+			</li>
+		);
+	}
+
+	render() {
+		return (
+			<div>
+				<QueryInlineHelpSearch query={ this.props.searchQuery } requesting={ this.props.isSearching } />
+				{ this.renderSearchResults() }
+			</div>
+		);
+	}
+}
+
+const mapStateToProps = ( state, ownProps ) => {
```

Comment:
nice-to-have simplification for your `mapStateToProps` in this PR might be:

```
const mapStateToProps = ( state, ownProps ) => ( {
  searchResults: …,
  isSearching: …,
} );
```

saves a bit of brackets and additional nesting but I guess comes down to a personal preference. up to you to ignore me :)